src/warc2zim/content_rewriting/generic.py

```diff
@@ -88,12 +135,36 @@ def rewrite_html(self, head_template: Template, css_insert: str | None):
             self.content_str
         )
 
+    @no_title
     def rewrite_css(self):
-        return ("", CssRewriter(self.url_rewriter).rewrite(self.content))
+        return CssRewriter(self.url_rewriter).rewrite(self.content)
 
-    def rewrite_js(self, opts):
-        rewriter = build_domain_specific_rewriter(self.path, self.url_rewriter)
-        return (
-            "",
-            rewriter.rewrite(self.content_str, opts),
-        )
+    @no_title
+    def rewrite_js(self, opts: dict[str, Any]):
+        ds_rules = get_ds_rules(self.orig_url_str)
+        rewriter = JsRewriter(self.url_rewriter, ds_rules)
+        return rewriter.rewrite(self.content.decode(), opts)
+
+    @no_title
+    def rewrite_jsonp(self):
+        content = self.content.decode()
+        match = JSONP_REGEX.match(content)
+        if not match:
+            return content
+
+        callback = extract_jsonp_callback(self.orig_url_str)
+
+        if not callback:
+            return content
+
+        return callback + match.group(1)
+
+    @no_title
+    def rewrite_json(self):
+        content = self.rewrite_jsonp()[1]
```

Comment:
Isn't it returning a string? I notice many methods not declaring a return type. We should.

Reply:
The method return a string, but it is decorated with `no_title` which makes it return a
tuple `("", rewritten_content)`.

Reply:
Then the type hint of the method is `str`, the decorator has its own type hint.

Reply:
Yes, I have added the type hint. But we still need to do the indexation `[1]` as the decorated `rewrite_jsonp` returns a tuple.